inference-engine/src/gna_plugin/gna_infer_request.hpp

```diff
@@ -69,6 +69,14 @@ class GNAInferRequest : public InferenceEngine::AsyncInferRequestInternal {
         // execute input pre-processing.
         execDataPreprocessing(_inputs);
         inferRequestIdx = plg->QueueInference(_inputs, _outputs);
+        // workaround to unblock callback-based flows
+        if (_callback) {
+            InferenceEngine::ResponseDesc resp;
+            auto infer_request = _publicInterface.lock();
```

Comment:
why do we need that? I think interface already locked since we are inside inferRequest_startasync method?
I would just call 

``` c++
_callback(this, Wait(-1));
```

Reply:
As discussed internally, it's needed because this is another hierarchy, and _publicInterface member contains a pointer to the original IInferRequest which needs to be passed to _callback.

Reply:
i still would use Wait() for wait - no mess with2 status code, res, and resp